spec/ddtrace/benchmark/support/benchmark_helper.rb

```diff
@@ -207,6 +268,11 @@ def server_runner
       conn.print AGENT_HTTP_RESPONSE
       conn.flush
 
+      # Read HTTP request to allow other side to have enough
+      # buffer write room. If we don't, the client won't be
+      # able to send the full request until the buffer is cleared.
+      conn.read(1 << 31 - 1)
```

Comment:
This is happening because, in the end-to-end test, we are trying to send very large payloads, due to the buffer constantly having close to 1000 items.

The network write buffer was getting full on the worker side, which would just block and not continue flushing until it timed out.